gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java

```diff
@@ -217,6 +252,16 @@ private Optional<String> diff(File existing, String upToDateContents) {
         }
     }
 
+    private File getSchemaLockfile() {
+        return getProject().file(SchemaVersionLockFile.LOCK_FILE);
+    }
+
+    private void ensureSchemaLockfileIsUpToDate(List<SchemaMigration> schemaMigrations) throws IOException {
+        File lockfile = getSchemaLockfile();
+        String upToDateContents = ObjectMappers.writeSchemaVersionsAsString(SchemaVersionLockFile.of(schemaMigrations));
+        ensureFileIsUpToDate(lockfile, upToDateContents);
```

Comment:
It feels a bit weird to me that `./gradlew writeProductDependenciesLocks` updates the schema lock file, given it has nothing to do with product dependencies. Could we split this up and maybe add something like `./gradlew writeSchemaMigrationLocks`?

Maybe we can parameterize the `CreateManifestTask#shouldWriteLocks` method to take take in the task name to check as an additional parameter and then pipe this information through from the respective `ensureXXXLockfileIsUpToDate` method?